tests/conftest.py

```diff
@@ -0,0 +1,64 @@
+import uuid
+
+import apypie
+import paramiko
+import pytest
+
+
+@pytest.fixture(scope="module")
+def ssh_config():
+    config = paramiko.SSHConfig.from_path('./.vagrant/ssh-config')
+    return config.lookup('quadlet')
+
+
+@pytest.fixture(scope="module")
+def foremanapi(ssh_config):
+    return apypie.ForemanApi(
+        uri=f'https://{ssh_config['hostname']}',
+        username='admin',
+        password='changeme',
+        verify_ssl=False,
+    )
+
+@pytest.fixture
+def organization(foremanapi):
+    org = foremanapi.create('organizations', {'name': str(uuid.uuid4())})
+    yield org
+    foremanapi.delete('organizations', org)
+
+@pytest.fixture
+def product(organization, foremanapi):
+    prod = foremanapi.create('products', {'name': str(uuid.uuid4()), 'organization_id': organization['id']})
+    yield prod
+    foremanapi.delete('products', prod)
+
+@pytest.fixture
+def yum_repository(product, organization, foremanapi):
+    repo = foremanapi.create('repositories', {'name': str(uuid.uuid4()), 'product_id': product['id'], 'content_type': 'yum', 'url': 'https://fixtures.pulpproject.org/rpm-no-comps/'})
+    yield repo
+    foremanapi.delete('repositories', repo)
+
+@pytest.fixture
+def file_repository(product, organization, foremanapi):
+    repo = foremanapi.create('repositories', {'name': str(uuid.uuid4()), 'product_id': product['id'], 'content_type': 'file', 'url': 'https://fixtures.pulpproject.org/file/'})
+    yield repo
+    foremanapi.delete('repositories', repo)
+
+@pytest.fixture
+def container_repository(product, organization, foremanapi):
+    repo = foremanapi.create('repositories', {'name': str(uuid.uuid4()), 'product_id': product['id'], 'content_type': 'docker', 'url': 'https://quay.io/', 'docker_upstream_name': 'foreman/busybox-test'})
+    yield repo
+    foremanapi.delete('repositories', repo)
+
+@pytest.fixture
+def lifecycle_environment(organization, foremanapi):
+    library = foremanapi.list('lifecycle_environments', 'name=Library', {'organization_id': organization['id']})[0]
```

Comment:
Same here perhaps?
```suggestion
def lifecycle_environment(organization, foremanapi, prior='Library'):
    library = foremanapi.list('lifecycle_environments', f'name={prior}', {'organization_id': organization['id']})[0]
```

Though perhaps it should just accept an instance where `None` is automatically resolve to the library:
```suggestion
def lifecycle_environment(organization, foremanapi, prior=None):
    if not prior:
        prior = foremanapi.list('lifecycle_environments', 'name=Library', {'organization_id': organization['id']})[0]
```

Which also makes me question what the Katello API does if no prior ID is given. It's not listed as required and mandating this, but would it make sense to find the library server side and make it an optional parameter?

Reply:
It does not find "Library" automatically and errors out:
https://github.com/Katello/katello/blob/db89c59031e34994805faaf3c6314b0e412b56c3/app/controllers/katello/api/v2/environments_controller.rb#L39
https://github.com/Katello/katello/blob/db89c59031e34994805faaf3c6314b0e412b56c3/app/controllers/katello/api/v2/environments_controller.rb#L172-L177

Reply:
I meant that last part as an RFE.